src/main/java/seedu/address/logic/commands/EditEventCommand.java

```diff
@@ -70,36 +70,69 @@ public EditEventCommand(Index patientIndex, Index eventIndex, Event eventToUpdat
     @Override
     public CommandResult execute(Model model) throws CommandException {
         List<Patient> lastShownList = model.getFilteredPatientList();
+        checkPatientIndex(lastShownList);
 
+        Patient patientToEditEvent = lastShownList.get(patientIndex.getZeroBased());
+        Set<Event> events = patientToEditEvent.getEvents();
+        checkEventIndex(events);
+
+        List<Event> eventList = new ArrayList<>(events);
+        editEvent(eventList);
+
+        Patient updatedPatient = createEditedPatient(patientToEditEvent, editPatientDescriptor);
+        updatePatientList(model, patientToEditEvent, updatedPatient);
+
+        if (events.contains(eventToUpdate)) {
+            return new CommandResult(String.format(MESSAGE_DUPLICATE, eventToUpdate.name, eventToUpdate.date,
+                    updatedPatient.getName(), patientIndex.getOneBased()));
+        }
```

Comment:
With reference to the previous comment, maybe shift this to Line 79, before `editEvent` is called

Reply:
Thanks for the advice!